### PR TITLE
Missing node.serviceAccount.annotations in values.yaml

### PR DESCRIPTION
**Pull request by netikras**
*Issue #, if available:*
564
*Description of changes:*
Added a default annotation for the SA.
ref.: https://github.com/awslabs/mountpoint-s3-csi-driver/issues/173#issuecomment-2209120237

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by unexge — APPROVED** [review]

File: docs/install.md [hunk 1]
```diff
@@ -114,6 +114,14 @@ helm upgrade --install aws-mountpoint-s3-csi-driver \
     aws-mountpoint-s3-csi-driver/aws-mountpoint-s3-csi-driver
 ```
 
+> [!NOTE]
+> For EKS users, you need to pass your Role ARN here if you're using IAM roles for service accounts:
+> 
+> $ helm upgrade --install aws-mountpoint-s3-csi-driver \
+>    --namespace kube-system \
+>    --set node.serviceAccount.annotations."eks\.amazonaws\.com/role-arn"="arn:aws:iam::account:role/csi-driver-role-name" \
+>    aws-mountpoint-s3-csi-driver/aws-mountpoint-s3-csi-driver
```
Comment (unexge) [inline docs/install.md:123]:
````suggestion
> ```bash
> $ helm upgrade --install aws-mountpoint-s3-csi-driver \
>    --namespace kube-system \
>    --set node.serviceAccount.annotations."eks\.amazonaws\.com/role-arn"="arn:aws:iam::account:role/csi-driver-role-name" \
>    aws-mountpoint-s3-csi-driver/aws-mountpoint-s3-csi-driver 
> ```
````

**Review by unexge — COMMENTED** [review]
Thanks for the update @netikras!

Could you please ensure the code block is properly formatted as a code block?
Screenshot 2024-07-10 at 16 51 55

Currently, it's not formatted as a code block and the escapes in the Helm command are filtered out and, Helm command fails to execute if you don't escape it

**Review by unexge — APPROVED** [review]

**Comment by unexge** [comment]
Thank you for the contribution again @netikras!

**Comment by netikras** [comment]
@unexge 
too many thanks for a contribution this small, if you asked me :smile:

**Pull request merged**